### PR TITLE
[Feature] suport SHOW AUTHENTICATION

### LINKED ISSUE #9855: Show grants in version 2.3 can't display authentication information
## Enhancement
In version 2.3, the `show grants` command can't display authentication information anymore. We need another way to check those information.

### PR DESCRIPTION
**Pull request by padmejin**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #9855

## Problem Summary(Required) ：

```sql
-- display authentication information for the user.
show authentication;
-- display all authentication information, require GRANT privilege
show all authentication;
-- display all authentication information. require GRANT privilege if user is not current login user.
show authentication for <user_name>;
```

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 46 / 82 (56.10%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[194]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|0|4|00.00%|[1014, 1015, 1016, 1019]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|1|7|14.29%|[290, 300, 301, 302, 303, 304]|
|:large_blue_circle:|com/starrocks/mysql/privilege/Auth.java|15|34|44.12%|[1446, 1447, 1450, 1451, 1453, 1454, 1457, 1458, 1459, 1460, 1461, 1463, 1465, 1479, 1485, 1486, 1491, 1497, 1523]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowStmtAnalyzer.java|7|11|63.64%|[600, 601, 602, 603]|
|:large_blue_circle:|com/starrocks/analysis/ShowAuthenticationStmt.java|16|18|88.89%|[35, 40]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|5|5|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 63 / 82 (76.83%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[194]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|1|7|14.29%|[290, 300, 301, 302, 303, 304]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowStmtAnalyzer.java|7|11|63.64%|[600, 601, 602, 603]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|3|4|75.00%|[1016]|
|:large_blue_circle:|com/starrocks/mysql/privilege/Auth.java|29|34|85.29%|[1451, 1479, 1485, 1486, 1523]|
|:large_blue_circle:|com/starrocks/analysis/ShowAuthenticationStmt.java|16|18|88.89%|[35, 40]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|5|5|100.00%|[]|

**Comment by padmejin** [comment]
run starrocks_admit_test

**Review by gengjun-git — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/analysis/ShowAuthenticationStmt.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+package com.starrocks.analysis;
+
+import com.starrocks.catalog.Column;
+import com.starrocks.catalog.ScalarType;
+import com.starrocks.qe.ShowResultSetMetaData;
+import com.starrocks.sql.ast.AstVisitor;
+
+public class ShowAuthenticationStmt extends ShowStmt {
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/analysis/ShowAuthenticationStmt.java:10]: authorization not authentication

Comment (padmejin) [reply fe/fe-core/src/main/java/com/starrocks/analysis/ShowAuthenticationStmt.java:10]: Authentication (from [Greek](https://en.wikipedia.org/wiki/Greek_language): αὐθεντικός authentikos, "real, genuine", from αὐθέντης authentes, "author") is the act of [proving](https://en.wikipedia.org/wiki/Proof_(truth)) an [assertion](https://en.wikipedia.org/wiki/Logical_assertion), such as the [identity](https://en.wikipedia.org/wiki/Digital_identity) of a computer system user.

**Review by chaoyli — APPROVED** [review]

**Comment by padmejin** [comment]
@Mergifyio backport branch-2.4 branch-2.3
¡Muchas gracias!

**Pull request merged**